lib/PEX.ts

```diff
@@ -315,108 +315,89 @@ export class PEX {
+        ? opts.basePresentationPayload['@context']
+        : [opts.basePresentationPayload['@context']]
+      : [];
+    if (!context.includes('https://www.w3.org/2018/credentials/v1')) {
+      context.push('https://www.w3.org/2018/credentials/v1');
+    }
 
-      const context = opts?.basePresentationPayload?.['@context']
-        ? Array.isArray(opts.basePresentationPayload['@context'])
-          ? opts.basePresentationPayload['@context']
-          : [opts.basePresentationPayload['@context']]
-        : [];
-      if (!context.includes('https://www.w3.org/2018/credentials/v1')) {
-        context.push('https://www.w3.org/2018/credentials/v1');
+    if (opts?.presentationSubmission) {
+      if (!type.includes('PresentationSubmission')) {
+        type.push('PresentationSubmission');
       }
-
-      if (opts?.presentationSubmission) {
-        if (!type.includes('PresentationSubmission')) {
-          type.push('PresentationSubmission');
-        }
-        if (!context.includes('https://identity.foundation/presentation-exchange/submission/v1')) {
-          context.push('https://identity.foundation/presentation-exchange/submission/v1');
-        }
+      if (!context.includes('https://identity.foundation/presentation-exchange/submission/v1')) {
+        context.push('https://identity.foundation/presentation-exchange/submission/v1');
       }
-      return {
+    }
+    const result: Array<IPresentation | PartialSdJwtDecodedVerifiableCredential> = [];
+    if (PEX.allowMultipleVCsPerPresentation(verifiableCredential)) {
+      result.push({
         ...opts?.basePresentationPayload,
         '@context': context,
         type,
         holder,
         ...(!!opts?.presentationSubmission && { presentation_submission: opts.presentationSubmission }),
         verifiableCredential,
-      };
+      });
+    } else {
+      verifiableCredential.forEach((vc) => {
+        if (CredentialMapper.isSdJwtDecodedCredential(vc)) {
+          result.push(vc as PartialSdJwtDecodedVerifiableCredential);
+        } else if (CredentialMapper.isSdJwtEncoded(vc)) {
+          const decoded = CredentialMapper.decodeVerifiableCredential(vc, opts?.hasher);
+          result.push(decoded as PartialSdJwtDecodedVerifiableCredential);
+        } else {
+          // This should be plain jwt only
+          result.push({
+            ...opts?.basePresentationPayload,
+            '@context': context,
+            type,
+            holder,
+            ...(!!opts?.presentationSubmission && { presentation_submission: opts.presentationSubmission }),
+            verifiableCredential: [vc],
+          });
+        }
+      });
     }
+    return result;
+  }
+
+  private static allowMultipleVCsPerPresentation(verifiableCredential: Array<OriginalVerifiableCredential>) {
+    return !verifiableCredential.some(
+      (c) => CredentialMapper.isJwtEncoded(c) || CredentialMapper.isSdJwtEncoded(c) || CredentialMapper.isSdJwtDecodedCredential(c),
```

Comment:
jwts allow multiple VCs per presentation; sd-jwts don't. But not with different holders. Please reread the ticket and either adjust the logic or add a big fat warning